### DISCUSSION TITLE
1.4.3 release task discussion

### DISCUSSION THREAD

**Discussion by innobead** [longhorn/longhorn#6304] [General]
Let's use this to discuss the 1.4.3 release QA process.  cc @longhorn/dev @longhorn/qa

**Comment by innobead** [comment]
@longhorn/qa RC2 is ready. Please help verify it and update the test result to https://github.com/longhorn/longhorn/wiki/Release-Test-Plan---1.4.3 and https://github.com/longhorn/longhorn/issues/6102.

@khushboo-rancher Please summarize the test result here, so we can base it on that to proceed with the GA build. Thanks.

**Comment by roger-ryao** [comment]
Hi @longhorn/dev and @longhorn/qa,
While verifying the Longhorn Chart using `Helm 2`, I encountered an error. Following @PhanLe1010's suggestion, we have created a ticket at https://github.com/longhorn/longhorn/issues/6318 to discuss this topic. The `Helm 3` test passed without any errors.

**Comment by yangchiu** [comment]
Not sure if we still need to take care `helm 2` since it's been end of life.

What I think we need to pay more attention to is `flux` (`gitops`) since it can reproduce the error about `admission-webhook` and `recovery-backend` that most users encountered during upgraded from v1.4.2 to v1.5.0. Like:
```
Warning  error   4m47s    helm-controller  Helm upgrade failed: deployments.apps "longhorn-recovery-backend" not found
```

But I don't know how to test our pre-release chart using flux since we didn't actually release it just like this user encountered:
https://github.com/longhorn/longhorn/issues/6246#issuecomment-1626402950
And they had to release their own customized chart by themselves to test:
https://github.com/longhorn/longhorn/issues/6246#issuecomment-1627360846

A more easier way for us to test is flux install helm chart based on git branch or commit, but how to do that is still unknown.

**Comment by khushboo-rancher** [comment]
About the [release test](https://github.com/longhorn/longhorn/wiki/Release-Test-Plan---1.4.3), things are looking fine, only two items are remaining (2 stage upgrade (arm) and uninstallation checks), they are in progress. We should be able to get the result in another 3-4 hours.

edit - Regression testing is completed.

**Comment by yangchiu** [comment]
Uninstallation checks for rc2 has been done and updated in the test plan, but somehow the change got reverted, I doubt the editing function would overwrite other's change. I've updated it again. FYI

**Comment by roger-ryao** [comment]
> About the [release test](https://github.com/longhorn/longhorn/wiki/Release-Test-Plan---1.4.3), things are looking fine, only two items are remaining (2 stage upgrade (arm) and uninstallation checks), they are in progress. We should be able to get the result in another 3-4 hours.

The 2-stage upgrade (arm) has been completed, and I have updated the test results.

**Comment by innobead** [comment]
@yangchiu Please create an issue for gitops deployment (install/upgrade) with most used solutions like flux/argocd.

**Comment by innobead** [comment]
@chriscchien how about ur part?

**Reply by chriscchien** [reply]
v1.4.3-rc2 regression test on SLES done, AMD64 2 failures, ARM64 3 failures. no new issue observed

**Reply by innobead** [reply]
> v1.4.3-rc2 regression test on SLES done, AMD64 2 failures, ARM64 3 failures. no new issue observed

Those failures are flaky or known issues? If known, the issues have been created?

**Reply by chriscchien** [reply]
1 flaky case [test_default_storage_class_syncup](https://ci.longhorn.io/job/public/job/v1.4.x/job/v1.4.x-longhorn-tests-sles-arm64/225/testReport/junit/tests/test_basic/test_default_storage_class_syncup/)  have a backport [PR](https://github.com/longhorn/longhorn-tests/pull/1470) in review.

The other 2 already have documented [here](https://github.com/longhorn/longhorn/issues/5460#issuecomment-1577914536)

- `test_inc_restoration_with_multiple_rebuild_and_expansion`
- `test_csi_mount_volume_online_expansion`

**Comment by innobead** [comment]
https://github.com/longhorn/longhorn/releases/tag/v1.4.3
